### PR TITLE
Enhancement: Schema / deployment form

### PR DESCRIPTION
**Pull request by znicholasbrown**
This PR makes some minor improvements to deployment forms and fixes a few long standing issues with schemas. One bigish change is the removal of `PSelect` in list properties; this is to work around some incompatibilities between the component and the intent of list properties (in particular duplicate values and ordering)

### CODE REVIEW

File: src/components/JsonInput.vue
Comment (pleek91) [inline src/components/JsonInput.vue:1]: This component is just for the format button basically?

File: src/components/SchemaFormInput.vue [hunk 1]
```diff
@@ -50,6 +50,13 @@
   grid
   gap-1
 }
+
+.schema-form-input__component { @apply
+  max-w-full
+  min-h-[2.5rem]
+  resize-y
+}
```
Comment (pleek91) [inline src/components/SchemaFormInput.vue:58]: This seems a little too specific to apply to all schema components. The max width makes sense but the min-h and resize?

File: src/components/JsonInput.vue [hunk 2]
```diff
@@ -17,8 +17,10 @@
   import { stringify } from '@/utilities/json'
 
   const props = defineProps<{
+    minLines?: number,
```
Comment (znicholasbrown) [inline src/components/JsonInput.vue:20]: I added these purely for the schema form `withPropsWithoutExcludedFactory` type, which only pulls props that are exposed from the component, not any that it wraps; there might be a better way to handle this

File: src/components/SchemaFormInput.vue (hunk 1 above)
Comment (znicholasbrown) [reply src/components/SchemaFormInput.vue:58]: I can target the code inputs instead, wdyt?

File: src/components/JsonInput.vue
Comment (znicholasbrown) [reply src/components/JsonInput.vue:1]: Yeah, I don't think the `PCodeInput` should support a format button since it only applies to 1 language right now

**Comment by znicholasbrown** [comment]
Form comparison:

Before:

https://user-images.githubusercontent.com/27291717/234671820-7173104c-0f0f-406c-abd0-a0fba4668af9.mov


After:

https://user-images.githubusercontent.com/27291717/234671847-2fbe1023-c93c-49a7-9e82-2eb7c7d00b48.mov

File: src/components/SchemaFormInput.vue (hunk 1 above)
Comment (pleek91) [reply src/components/SchemaFormInput.vue]: I'm not sure, this just seems overly generic for this. Could be wrong 🤷

File: src/components/JsonInput.vue (hunk 2 above)
Comment (pleek91) [reply src/components/JsonInput.vue:20]: this makes sense to me. This is a general problem that soon we'll be able to fix (in vue 3.3) by making props their own types and exporting them from the component itself (cannot export multiple from a sfc today). Then in a component like this you could import the type and add it to your props (cannot use an imported type for defineProps today). Both fixed in 3.3

File: src/components/SchemaFormInput.vue (hunk 1 above)
Comment (znicholasbrown) [reply src/components/SchemaFormInput.vue]: I added a more targeted selector for now cause I agree that was too generic. That'll let us close some tickets around resizing these inputs too

**Review by pleek91 — APPROVED** [review]

**Pull request merged**